tools/audio_common.py

```diff
@@ -0,0 +1,1193 @@
+#!/usr/bin/env python3
+
+import xml.etree.ElementTree as XmlTree
+import struct
+import math
+
+#Common Class Definitions
+class StructPackSpecs:
+	def __init__(self, archLE=False, arch64=False):
+		self.byte_order_char = ">"
+		if archLE:
+			self.byte_order_char = "<"
+		self.is_64 = arch64;
+		
+	def pointerSize(self):
+		if self.is_64:
+			return 8
+		else:
+			return 4
+		
+	def pointerPaddingSize(self):
+		if self.is_64:
+			return 4
+		else:
+			return 0		
+		
+	def genPackString(self, base_string):
```

Comment:
Figured it out. I'm curious what the purpose of P and X are supposed to be.

Reply:
Yup, so I basically copied the method of the sm64 assembly script for adding Little-Endian/64-bit arch options. P represents pointer/offset (which would be 8 instead of 4 bytes if 64 bit) and X represents a placeholder for padding (since usually 64-bit compilers insist on ensuring pointers are aligned to 8 bytes which given the N64 struct layout means padding would probably be added between fields)

Reply:
Gotcha, that makes more sense.  So P represents a pointer and X represents alignment padding.